### PR TITLE
minor: [BLUEP-570] script for crawling events in azure

### PR DESCRIPTION
**Pull request by usmanklodhi**
https://k6scope.atlassian.net/browse/BLUEP-570

### CODE REVIEW

**Comment by MDanialSaleem** [comment]
The resource for event crawl and data crawl should be set up as part of a single terrform module.

File: azure/kscope_crawl/main.tf [hunk 1]
```diff
@@ -45,8 +46,67 @@ resource "azuread_application" "example" {
...
+resource "azurerm_storage_account" "example" {
+  name                     = "k6scopemystorageaccount"
+  resource_group_name      = azurerm_resource_group.example.name
+  location                 = azurerm_resource_group.example.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_storage_container" "example" {
+  name                  = "k6scopemycontainer"
```
Comment (waleedakramkhan) [inline azure/kscope_crawl/main.tf:63]:
```suggestion
  name                  = "kaleidoscopeactivitylogscontainer"
```

File: azure/kscope_crawl/main.tf [hunk 2]
```diff
@@ -45,8 +46,67 @@ resource "azuread_application" "example" {
...
+  storage_account_name  = azurerm_storage_account.example.name
+  container_access_type = "private"
+}
+
+resource "azurerm_storage_queue" "example" {
+  name                 = "kscopemyqueue"
```
Comment (waleedakramkhan) [inline azure/kscope_crawl/main.tf:69]:
```suggestion
  name                 = "kaleidoscopeactivitylogsqueue"
```

File: azure/kscope_crawl/main.tf [hunk 3]
```diff
@@ -45,8 +46,67 @@ resource "azuread_application" "example" {
...
+  storage_account_name = azurerm_storage_account.example.name
+}
+
+locals {
+  resourceId = "${azurerm_storage_account.example.id}/blobServices/default"
+}
+
+resource "azurerm_monitor_diagnostic_setting" "storage_account_logs" {
```
Comment (waleedakramkhan) [inline azure/kscope_crawl/main.tf:77]: can we do this for a subscription?

**Review by waleedakramkhan — CHANGES_REQUESTED** [review]
minor comments

File: azure/kscope_crawl/main.tf [hunk 4]
```diff
@@ -45,8 +46,67 @@ resource "azuread_application" "example" {
...
+  resource_group_name      = azurerm_resource_group.example.name
+  location                 = azurerm_resource_group.example.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_storage_container" "example" {
+  name                  = "k6scopemycontainer"
+  storage_account_name  = azurerm_storage_account.example.name
+  container_access_type = "private"
```
[the remark this answers is not in the document]
Comment (usmanklodhi) [reply azure/kscope_crawl/main.tf:65]: Applications outside the storage account's context will not have access to the container. By default, only the storage account owner and authorised users have access to the container.

File: azure/kscope_crawl/main.tf (hunk 3 above)
Comment (usmanklodhi) [reply azure/kscope_crawl/main.tf:77]: Diagnostic settings are not available at subscription level. I've checked thoroughly.

File: azure/kscope_crawl/main.tf [hunk 5]
```diff
@@ -45,8 +46,67 @@ resource "azuread_application" "example" {
   }
 }
 
+resource "azurerm_resource_group" "example" {
+  name     = "storage-rg"
+  location = "eastus"
```
[the remark this answers is not in the document]
Comment (usmanklodhi) [reply azure/kscope_crawl/main.tf]: Name and location for resource group are now being sent as inputs.

File: azure/kscope_crawl/main.tf (hunk 4 above)
[the remark this answers is not in the document]
Comment (waleedakramkhan) [reply azure/kscope_crawl/main.tf:65]: yes i get that but is our app also authorised to access this storage account and its blobs?

[the remark this answers is not in the document]
Comment (usmanklodhi) [reply azure/kscope_crawl/main.tf:65]: I think no, should i set the `container_access_type` to public?

[the remark this answers is not in the document]
Comment (waleedakramkhan) [reply azure/kscope_crawl/main.tf:65]: i think so, yes. otherwise we will get a access denied

[the remark this answers is not in the document]
Comment (usmanklodhi) [reply azure/kscope_crawl/main.tf]: The `container_access_type` is now container. It allows public access to the container and all the blobs within it. This means that anyone with the container URL can list and access all the blobs stored within the container.

**Review by waleedakramkhan — APPROVED** [review]
🚀

**Pull request merged**